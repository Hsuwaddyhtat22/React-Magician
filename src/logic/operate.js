import Big from 'big.js';

export default function operate(numberOne, numberTwo, operation) {
  const one = Big(numberOne);
  const two = Big(numberTwo);

  if (operation === '+') {
    return one.plus(two).toString();
  }

  if (operation === '-') {
    return one.minus(two).toString();
  }

  if (operation === 'x' || operation === '*') {
    return one.times(two).toString();
  }

  if (operation === 'รท' || operation === '/') {
    if (two.eq(0)) { // Check if the denominator is zero
      throw new Error("Can't divide by 0.");
    }
    return one.div(two).toString();
  }

  if (operation === '%') {
    if (two.eq(0)) { // Check if the denominator is zero
      throw new Error("Can't find modulo as can't divide by 0.");
    }
    return one.mod(two).toString();
  }

  throw new Error(`Unknown operation '${operation}'`);
}
