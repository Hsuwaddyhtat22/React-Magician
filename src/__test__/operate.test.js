import operate from '../logic/operate';

test('adds two numbers', () => {
    expect(operate('2', '3', '+')).toMatchSnapshot();
});

test('subtracts two numbers', () => {
    expect(operate('5', '3', '-')).toMatchSnapshot();
});

test('multiplies two numbers', () => {
    expect(operate('4', '2', '*')).toMatchSnapshot();
});

test('divides two numbers', () => {
    expect(operate('6', '2', '/')).toMatchSnapshot();
});

test('throws an error when dividing by zero', () => {
    expect(() => operate('10', '0', '/')).toThrowErrorMatchingSnapshot();
});

test('finds modulo of two numbers', () => {
    expect(operate('10', '3', '%')).toMatchSnapshot();
});

test('throws an error when finding modulo by zero', () => {
    expect(() => operate('10', '0', '%')).toThrowErrorMatchingSnapshot();
});

test('throws an error for an unknown operation', () => {
    expect(() => operate('10', '5', '^')).toThrowErrorMatchingSnapshot();
});