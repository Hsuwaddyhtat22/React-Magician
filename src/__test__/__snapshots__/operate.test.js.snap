// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds two numbers 1`] = `"5"`;

exports[`divides two numbers 1`] = `"3"`;

exports[`finds modulo of two numbers 1`] = `"1"`;

exports[`multiplies two numbers 1`] = `"8"`;

exports[`subtracts two numbers 1`] = `"2"`;

exports[`throws an error for an unknown operation 1`] = `"Unknown operation '^'"`;

exports[`throws an error when dividing by zero 1`] = `"Can't divide by 0."`;

exports[`throws an error when finding modulo by zero 1`] = `"Can't find modulo as can't divide by 0."`;
